### ISSUE TITLE
[CT-780] [Bug] ... is not of type 'string'

### ISSUE THREAD

**Issue by pascalvogler** [dbt-labs/dbt-core#5416]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hello 

**tldr: dbt transforms yml strings to int values, which causes dbt to break.**

I have a source yml file with the following source (ids changed):

- name: "12345678"
    database: my-bigquery-project-1234
    tables:
      - name: ga_sessions
        identifier: "ga_sessions_*"

for me this works without a problem. However when a co-worker tried to install dbt and run a command the following error message occured:

`Invalid sources config given in [path_to_this_source_file.yml] @ sources: ['name': 12345678, 'database': 'my-bigquery-project-1234', 'tables': [{'name': 'ga_sessions', 'identifier': 'ga_sessions_*'}]] - at path ['name']: 12345678 is not of type 'string'`

how can we fix this? Is this a bug somehow, caused by some other package maybe? Or do you have a hint to solve the issue?

### Expected Behavior

The name of the source dataset should be interpreted as '12345678' (type string) as indicated by the quotes in the yml file, and not as integer (which also causes the error)

### Steps To Reproduce

running a simple dbt command with a source file (see in "Current Behavior") (python 3.10) 
I don't really know how to reproduce the error, since it works for me personally, but not for my colleague.

### Relevant log output

```shell
`Invalid sources config given in [path_to_this_source_file.yml] @ sources: ['name': 12345678, 'database': 'my-bigquery-project-1234', 'tables': [{'name': 'ga_sessions', 'identifier': 'ga_sessions_*'}]] - at path ['name']: 12345678 is not of type 'string'`
```

### Environment

```markdown
- OS: mac
- Python: 3.10
- dbt: core: 1.1.1 (latest), bq: 1.1.0 (latest)

These are the packages when pip list'ing:

agate                    1.6.3
attrs                    21.4.0
Babel                    2.10.3
cachetools               5.2.0
certifi                  2022.6.15
cffi                     1.15.0
charset-normalizer       2.0.12
click                    8.1.2
colorama                 0.4.4
dbt-bigquery             1.1.0
dbt-core                 1.1.1
dbt-extractor            0.4.1
Flask                    2.1.1
future                   0.18.2
gcloud                   0.18.3
google-api-core          2.8.2
google-auth              2.8.0
google-cloud-bigquery    2.34.4
google-cloud-core        2.3.1
google-crc32c            1.3.0
google-resumable-media   2.3.3
googleapis-common-protos 1.56.3
grpcio                   1.47.0
grpcio-status            1.47.0
gunicorn                 20.1.0
hologram                 0.0.15
httplib2                 0.20.4
idna                     3.3
importlib-metadata       4.11.3
isodate                  0.6.1
itsdangerous             2.1.2
Jinja2                   2.11.0
jsonschema               3.2.0
leather                  0.3.4
Logbook                  1.5.3
MarkupSafe               2.0.1
mashumaro                2.9
minimal-snowplow-tracker 0.0.2
msgpack                  1.0.4
networkx                 2.8.3
oauth2client             4.1.3
packaging                21.3
parsedatetime            2.4
pip                      21.3.1
proto-plus               1.20.6
protobuf                 3.20.1
pyasn1                   0.4.8
pyasn1-modules           0.2.8
pycparser                2.21
pyparsing                3.0.9
pyrsistent               0.18.1
python-dateutil          2.8.2
python-slugify           6.1.2
pytimeparse              1.1.8
pytz                     2022.1
PyYAML                   6.0
requests                 2.28.0
rsa                      4.8
ruamel.yaml              0.17.21
ruamel.yaml.clib         0.2.6
setuptools               60.2.0
six                      1.16.0
sqlparse                 0.4.2
text-unidecode           1.3
typing_extensions        4.2.0
urllib3                  1.26.9
Werkzeug                 2.1.0
wheel                    0.37.1
zipp                     3.7.0
```

### What database are you using dbt with?

bigquery

**Comment by pascalvogler** [comment]
could be fixed. it was the package MarkupSafe, which was not Version 2.1.1 . 
although when installing DBT the following message popped up:
`dbt-core 1.1.1 requires MarkupSafe<2.1,>=0.23, but you have markupsafe 2.1.1 which is incompatible.`

so in the end "ignoring" this error message solved the issue
